File: 백준/Silver/25206. 너의 평점은/너의 평점은.java
import java.io.*;
import java.util.HashMap;
import java.util.Map;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) throws IOException {

        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        Map<String, Double> grade_std = new HashMap<>();

        grade_std.put("A+", 4.5);
        grade_std.put("A0", 4.0);
        grade_std.put("B+", 3.5);
        grade_std.put("B0", 3.0);
        grade_std.put("C+", 2.5);
        grade_std.put("C0", 2.0);
        grade_std.put("D+", 1.5);
        grade_std.put("D0", 1.0);
        grade_std.put("F", 0.0);

        double total_credit = 0;
        double major_grade = 0;


        for (int i = 0; i < 20; i++) {

            StringTokenizer st = new StringTokenizer(br.readLine(), " ");

            String object = st.nextToken();
            double credit = Double.parseDouble(st.nextToken());
            String str_grade = st.nextToken();

            if (str_grade.equals("P")){
                continue;
            }
            double grade = grade_std.get(str_grade);

                major_grade += credit * grade;
                total_credit += credit;
                
        }
        double answer = major_grade/total_credit;
        String formated_answer = String.format("%.6f",answer);

        sb.append(formated_answer);
        System.out.println(sb);
    }
           }

File: 백준/Silver/25206. 너의 평점은/너의 평점은.py
import sys

grade_std = {"A+": 4.5, "A0": 4.0, "B+": 3.5, "B0": 3.0, "C+": 2.5, "C0": 2.0, "D+": 1.5, "D0": 1.0, "F":0.0}

total_credit = 0
major_grade = 0

for i in range(20):
    object_name, credit, grade = (sys.stdin.readline().split())
    if grade == "P":
        continue
    major_grade += float(credit) * grade_std[grade]
    total_credit += float(credit)

answer = major_grade/total_credit
print("%.6f" %answer)